software/pairtools/dedup/main.nf

```diff
@@ -0,0 +1,41 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process PAIRTOOLS_DEDUP {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::pairtools=0.3.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/pairtools:0.3.0--py37hb9c2fc3_5"
+    } else {
+        container "quay.io/biocontainers/pairtools:0.3.0--py37hb9c2fc3_5"
+    }
+
+    input:
+    tuple val(meta), path(input)
+
+    output:
+    tuple val(meta), path("*.dedup.pairs.gz")  , emit: pairs
+    tuple val(meta), path("*.dedup.pairs.stat"), emit: stat
+    path "*.version.txt"                       , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    pairtools dedup \\
+        $options.args \\
+        -o ${prefix}.dedup.pairs.gz \\
+        --output-stats ${prefix}.dedup.pairs.stat \\
+        ${input}
```

Comment:
```suggestion
        -o ${prefix}.gz \\
        --output-stats ${prefix}.stat \\
        $input
```

Reply:
We should try to minimise customising the filename as much as possible so it can be as flexible as possible for developers to name the output files.